### PR TITLE
Social media website - Milestone: friendships v1

### PR DESCRIPTION
**Pull request by Emmyn5600**
## On this pull request I implemented the following features:

* ### Updated Readme file
* ### Created a model with associations and all requested features for friendships.
* ### I styled the Add friendship button 
* ### Make User account visible at the top of Navbar with his or her information.

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -23,7 +23,12 @@ default: &default
```
Comment (mricanho) [inline config/database.yml:27]: Change this file to work only with default, without username and password

**Review by mricanho — CHANGES_REQUESTED** [review]
# 🔴  Changes Required 🔴  

Hello @Emmyn5600, you did a good job on this milestone🔥, just a few changes are required:

- [x]  **You need to change the .ruby-version file from 2.7.0 to 2.7.2** 
- [x] **Change the database.yml to work with default without user and password**
- [x] **The button Add Friend must disappear or change in any way once you click on it, because, you know, you only Add a Friend once until the other person accept or reject the friendship**
- [x] **You don't need a button to Add Friend if it's your own profile, right now you have the button active for your own profile**
- [x] **When a friendship is to send, the person who receives it needs a notification about it**
- [x] **Your Add Friend action controller, right now have a routing problem that prevents that something happens once you click it, please fix it**
- [x] **User can reject a friendship request, please add this perk**
- [x] **Add unit tests for your methods and integration tests**

I know it's a lot, but got this!

tuesday


You can contact me on slack if you have any questions just search for me as @ Miguel Ricaño.

### 💻 👩🏽‍💻  Happy Coding!👨🏽‍💻  🖥️

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:wrench::hammer::full_moon:
**Hi Emmy!
Great job so far:clap::clap:. However, we just need to make a few changes to get the project approved:muscle:.**

**To improve**

- [ ] I get this [error](https://imgur.com/a/JUbyqeb) when I run `bundle exec rspec`. Check if the path for the `require_relative` is the correct path for both the implemented `friendship` and `user` model. And kindly add more tests. Both unit and integration test.

**Optional**

- [ ] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)). Currently, it includes every registered users in the app. Instead of the current logged-in `user` and his/her `friends`. It will be nice to fix this.

Make the above-required changes and when you are ready, request another review.

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hello @Emmyn5600 , kindly fix the mentioned issues, for me the rspec is still failing with the following error on screenshot
Screenshot from 2021-05-12 12-45-09

**Comment by Emmyn5600** [comment]
You can check, now it is working fine.

**Review by ElijahTobs — APPROVED** [review]
# Status: Approved! :tada: 

## Congratulations @Emmyn5600  

### Good job implementing the change as requested by the previous TSE.

## OPTIONAL 
* [ ] Your posts timeline should include posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)). Currently, it includes posts from other users who are not friends.


## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Pull request merged**